tools/windows_buildenv.bat

```diff
@@ -97,3 +102,38 @@ EXIT /B 0
     SET BUILDENV_NAME=!BUILDENV_NAME:CONFIGURATION=%CONFIGURATION%!
     SET RETVAL=%BUILDENV_NAME%
     GOTO :EOF
+
+:GENERATE_CMakeSettings_JSON
+REM Generate CMakeSettings.json which is read by MS Visual Studio to determine the supported CMake build environments
+    SET CMakeSettings="%MIXXX_ROOT%\CMakeSettings.json"
+    IF EXIST %CMakeSettings% del /f /q %CMakeSettings%
+    >>%CMakeSettings% echo {
+    >>%CMakeSettings% echo   "configurations": [
+    >>%CMakeSettings% echo     {
+    >>%CMakeSettings% echo       "buildCommandArgs": "",
+    >>%CMakeSettings% echo       "buildRoot": "${projectDir}\\cmake_build",
+    >>%CMakeSettings% echo       "cmakeCommandArgs": "-DDEBUG_ASSERTIONS_FATAL=ON -DHSS1394=ON -DKEYFINDER=OFF -DLOCALECOMPARE=ON -DMAD=ON -DMEDIAFOUNDATION=ON -DSTATIC_DEPS=ON -DBATTERY=ON -DBROADCAST=ON -DBULK=ON -DHID=ON -DLILV=ON -DOPUS=ON -DQTKEYCHAIN=ON -DVINYLCONTROL=ON",
+    >>%CMakeSettings% echo       "configurationType": "Release",
+    >>%CMakeSettings% echo       "ctestCommandArgs": "",
+    >>%CMakeSettings% echo       "generator": "Ninja",
+    >>%CMakeSettings% echo       "inheritEnvironments": [ "msvc_!PLATFORM!_!PLATFORM!" ],
+    >>%CMakeSettings% echo       "installRoot": "${projectDir}\\cmake_dist",
+    >>%CMakeSettings% echo       "intelliSenseMode": "windows-msvc-!PLATFORM!",
+    >>%CMakeSettings% echo       "name": "!PLATFORM!-!CONFIGURATION!",
+    >>%CMakeSettings% echo       "variables": [
+    >>%CMakeSettings% echo         {
+    >>%CMakeSettings% echo           "name": "CMAKE_PREFIX_PATH",
+    >>%CMakeSettings% echo           "type": "STRING",
+    REM Replaces all \ by \\ in CMAKE_PREFIX_PATH
+    >>%CMakeSettings% echo           "value": "!CMAKE_PREFIX_PATH:\=\\!"
+    >>%CMakeSettings% echo         },
+    >>%CMakeSettings% echo         {
+    >>%CMakeSettings% echo           "name": "CMAKE_INTERPROCEDURAL_OPTIMIZATION",
+    >>%CMakeSettings% echo           "type": "BOOL",
+    >>%CMakeSettings% echo           "value": "FALSE"
+    >>%CMakeSettings% echo         }
+    >>%CMakeSettings% echo       ]
+    >>%CMakeSettings% echo     }
+    >>%CMakeSettings% echo   ]
+    >>%CMakeSettings% echo }
+    GOTO :EOF
```

Comment:
Missing newline at EOF. Please use our pre-commit hooks to fix these kind of issue automatically.